### ISSUE TITLE
Align `Cell` model with the catalogue

### ISSUE THREAD

**Issue by gpelouze** [NaaVRE/NaaVRE-containerizer-service#10]
In the [catalogue](https://github.com/NaaVRE/NaaVRE-catalogue-service), we tweaked a couple of things about the cell model (see [implementation](https://github.com/NaaVRE/NaaVRE-catalogue-service/blob/main/app/workflow_cells/models.py) and [sample serialization](https://gist.github.com/gpelouze/ad8d467f7124fc128c78a6bf5df78127)). 

@skoulouzis, does it make sense to modify the model used in the containerizer service to match the one in the catalogue?

If we keep the current model in the containerizer service, we will need to translate the cell representation to match the catalogue's model in the frontend (after receiving the response from `NaaVRE-containerizer-service/containerize` and posting the cell to `NaaVRE-catalogue-service/workflow-cells/`). This is doable, but we might create confusion by having two formulations of the cell model across different services. 

---

Notable changes:

- `types` is removed and its values merged with `inputs`, `outputs`, `params` or `secrets`. Eg.

```json
{
  ...
  "inputs": [
    "test_input"
  ],
  "types": {
    "test_input": "float"
  },
}
```

becomes

```json
{
  ...
  "inputs": [
    {
      "name": "test_input",
      "type": "float"
    }
  ]
}
```

- `param_values` is removed and its values merged with `params`. Eg.

```json
{
  ...
  "params": [
    "param_test"
  ],
  "types": {
    "param_test": "str"
  },
  "param_values": {
    "param_test": "foo"
}
```

becomes

```json
{
  ...
  "params": [
    {
      "name": "param_test",
      "type": "str",
      "default_value": "\"foo\""
    }
  ],
}
```

- renamed: `base_image` → `base_container_image`
- added: `description`, `virtual_lab` (probably should be added in the frontend), `container_image` and `source_url` (the source folder on GitHub)
- removed: `node_id`, `task_name`, `all_inputs`, `base_image`, `notebook_dict` and `original_source` are removed
- `chart_obj` is not in the catalogue model, but it would be nice to still return it for now. We don't have TS code to generate it in the frontend, but we have the Py code to generate it in the backend. 
- (`owner`, `created`, `modified` are automatically added upon creation in the catalogue)

**Comment by skoulouzis** [comment]
@gpelouze yes, that makes sense. We should align all models with the one in the catalogue

**Comment by gpelouze** [comment]
@skoulouzis thanks, I see you added most of it. I updated the frontend to use the same model. 

However, I’m missing the following for the catalogue:
- `container_image`: Containerized cell image (example: ghcr.io/me/my-naavre-cells/my-cell-1:49c621b)
- `source_url`: URL of the folder on GitHub containing the sources of the image (example: https://github.com/me/my-NaaVRE-cells/tree/2934de123c74316dc45fe84d340a7ca6914b8bc1/my-cell-1)

Is it possible to retrieve them when creating the cell in GitHub and add them to the response of `/containerize`?

For now, I have added some dummy values but we should remove them in the future: https://github.com/NaaVRE/NaaVRE-containerizer-jupyterlab/issues/8

**Comment by gpelouze** [comment]
Here is an example response from the `POST /containerize`:

```json
{
  "workflow_id": "4d91a242-524e-4b1e-b9aa-e425b3ca0818",
  "dispatched_github_workflow": true,
  "image_version": "5b46f6c",
  "workflow_url": "https://github.com/QCDIS/NaaVRE-cells-test-3/actions/runs/12909341740/job/35996802690"
}
```

We should make the following changes:

- Add `container_image`, eg. `ghcr.io/qcdis/naavre-cells-test-3/untitled-test-user-2:5b46f6c`. This is probably easy, because the image name is generated alongside the current `image_version`
- Remove `image_version` (it is not used by the frontend and will be part of `container_image`)
- Add `source_url`, eg. `https://github.com/QCDIS/NaaVRE-cells-test-3/tree/2d51d063d12f4832f5daffde270575a4f3a68860/untitled-test-user-2`. This could be trickier, because we several commits are made when creating or updating a cell (one per file?). The last one should be used. When containerizing the cell for the above example, two commits were created (QCDIS/NaaVRE-cells-test-3@a7a13f96badbdd0613c7371285847136bfaa71a4 and QCDIS/NaaVRE-cells-test-3@2d51d063d12f4832f5daffde270575a4f3a68860), the last one should be used.

**Comment by skoulouzis** [comment]
Added new issue: https://github.com/NaaVRE/NaaVRE-containerizer-service/issues/27
